src/transformers/models/mgp_str/processing_mgp_str.py

```diff
@@ -0,0 +1,185 @@
+        [`~MGPSTRTokenizer.__call__`]. Please refer to the doctsring of the above method for more information.
+        """
+        images = kwargs.pop("images", None)
+        if len(args) > 0:
+            images = args[0]
+            args = args[1:]
+
+        if images is None:
+            raise ValueError("You need to specify an `images` input to process.")
+
+        inputs = self.feature_extractor(images, *args, **kwargs)
+        return inputs
+
+    def batch_decode(self, sequences):
+        """
+        Convert a list of lists of token ids into a list of strings by calling decode.
+
+        Args:
+            sequences (`torch.Tensor`):
+                List of tokenized input ids.
+
+        Returns:
+            `Dict[str, any]`: Dictionary of all the outputs of the decoded results.
+                generated_text (`List[str]`): The final results after fusion of char, bpe, and wp. scores
+                (`List[float]`): The final scores after fusion of char, bpe, and wp. char_preds (`List[str]`): The list
+                of character decoded sentences. bpe_preds (`List[str]`): The list of bpe decoded sentences. wp_preds
+                (`List[str]`): The list of wp decoded sentences.
+
+        This method forwards all its arguments to PreTrainedTokenizer's [`~PreTrainedTokenizer.batch_decode`]. Please
+        refer to the docstring of this method for more information.
+        """
+        char_preds, char_preds_softmax, bpe_preds, bpe_preds_softmax, wp_preds, wp_preds_softmax = sequences
+        batch_size = char_preds.size(0)
+
+        char_strs, char_scores = self._decode_helper(char_preds, char_preds_softmax, "char")
+        bpe_strs, bpe_scores = self._decode_helper(bpe_preds, bpe_preds_softmax, "bpe")
+        wp_strs, wp_scores = self._decode_helper(wp_preds, wp_preds_softmax, "wp")
+
+        final_strs = []
+        final_scores = []
+        for i in range(batch_size):
+            scores = [char_scores[i], bpe_scores[i], wp_scores[i]]
+            strs = [char_strs[i], bpe_strs[i], wp_strs[i]]
+            max_score_index = scores.index(max(scores))
+            final_strs.append(strs[max_score_index])
+            final_scores.append(scores[max_score_index])
+
+        out = {}
+        out["generated_text"] = final_strs
+        out["scores"] = final_scores
+        out["char_preds"] = char_strs
+        out["bpe_preds"] = bpe_strs
+        out["wp_preds"] = wp_strs
+        return out
+
+    def _decode_helper(self, pred_logits, pred_softmax, decode_type):
+        """
+        Convert a list of lists of bpe token ids into a list of strings by calling bpe tokenizer.
+
+        Args:
+            pred_logits (`torch.Tensor`):
+                List of model prediction logits.
+            pred_softmax (`torch.Tensor`):
+                List of model prediction softmax.
+            tyope (`str`):
+                Type of model prediction. Must be one of ['char', 'bpe', 'wp'].
```

Comment:
I would instead have this as an enum c.f. [the ExplicitEnum](https://github.com/huggingface/transformers/blob/0db5d911fc94604f9568b4b212e005ec4600d157/src/transformers/utils/generic.py#L309) class used in this repo